dask/dataframe/groupby.py

```diff
@@ -174,15 +174,51 @@ def factory(func, column):
     return factory
 
 
-def _aggregate_meta(df, index, levels, chunk_funcs, agg_funcs, finalizers):
-    stage1 = _groupby_apply_funcs(df, funcs=chunk_funcs, by=index)
-    stage2 = _groupby_apply_funcs(stage1, funcs=agg_funcs, level=levels)
-    return _agg_finalize(stage2, finalizers)
-
-
 def _normalize_spec(spec, non_group_columns):
     """
     Return a list of ``(result_column, func, input_column)`` tuples.
+
+    Spec can be
+
+    - a function
+    - a list of functions
+    - a dictionary that maps input-columns to functions
+    - a dictionary that maps input-columns to a lists of functions
+    - a dictionary that maps input-columns to a dictionaries that map
+      output-columns to functions.
+
+    The non-group columns are a list of all column names that are not used in
+    the groupby operation.
+
+    Usually, the result columns are be a mutli-level names, returned as a
+    tuples. If only a single function is supplied or dictionary mapping columns
+    to single functions, simple names are returned as strings (see the first
+    two examples below).
+
+    Examples
+    --------
+
+        >>> _normalize_spec('mean', ['a', 'b', 'c'])
+        [('a', 'mean', 'a'), ('b', 'mean', 'b'), ('c', 'mean', 'c')]
```

Comment:
These should be unindented to line up with the normal text on the left.  We should also drop the newline after the `----`.

http://dask.readthedocs.io/en/latest/develop.html#docstrings